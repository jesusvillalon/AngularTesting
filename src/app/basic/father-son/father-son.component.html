<h3>Hijo</h3>
<code class="mt-2 fs-5 fw-semibold">{{client | json }}</code>

<br><br>

<button class="btn btn-outline-dark"
*ngIf="client"
(click)="onChange(5)"s
data-test="btn-id"
>Cambiar ID</button>

<button class="btn btn-outline-danger"
*ngIf="client"
(click)="onDelete()"
data-test="btn-delete"
>Borrar cliente</button>
