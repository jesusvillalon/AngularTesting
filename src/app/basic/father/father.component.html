<h1>Padre</h1>
<button class="btn btn-dark"
(click)="onSetClient('JesÃºs')"
>Cargar cliente</button>
<br><br>

<code class="mt-2 fs-5 fw-semibold">{{client | json }}</code>

<hr>

<app-father-son [client]="client"
(onDeleteClient)="client = undefined"
(onClientUpdated)="client = $event"
  />
