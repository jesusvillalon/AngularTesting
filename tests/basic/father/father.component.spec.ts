import { ComponentFixture, TestBed } from '@angular/core/testing';

import { FatherComponent } from '../../../src/app/basic/father/father.component';
import { FatherSonComponent } from '../../../src/app/basic/father-son/father-son.component';
import { By } from '@angular/platform-browser';

describe('FatherComponent', () => {
  let component: FatherComponent;
  let fixture: ComponentFixture<FatherComponent>;
  let compiled: HTMLElement;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [FatherComponent, FatherSonComponent]
    });
    fixture = TestBed.createComponent(FatherComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe hacer match con el snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  test('debe de establecer el cliente con el nombre indicado', () => {
    component.onSetClient('Pedro');
    fixture.detectChanges();

    const codeDiv = compiled.querySelector('.mt-2');
    expect(codeDiv?.textContent).toContain('"name"');
    expect(codeDiv?.textContent).toContain('Pedro');
  });

  test('debe de borrar el cliente si se emite onDeleteClient (hijo)', () => {
    component.client = {id: 1, name: "Eduardo"};
    fixture.detectChanges();

    const sonDebugElement = fixture.debugElement.query(By.directive(FatherSonComponent));
    const sonComponent:FatherSonComponent = sonDebugElement.componentInstance;

    sonComponent.onDeleteClient.emit();
    expect(component.client).toBe(undefined)

  });

  test('deb3e de actualizar el cliente onClientUpdated', () => {
    component.client = {id: 1, name: "Eduardo"};
    fixture.detectChanges();

    const sonDebugElement = fixture.debugElement.query(By.directive(FatherSonComponent));
    const sonComponent:FatherSonComponent = sonDebugElement.componentInstance;

    sonComponent.onClientUpdated.emit({ id: 10, name: 'Pedro'});
    expect(component.client).toEqual({id: 10, name: 'Pedro'});

    // Usar toBe cuando algo es igual a un string, number o booleano.
    // Usar toEqual cuando un objeto sea igual a otro objeto.

  });




});
