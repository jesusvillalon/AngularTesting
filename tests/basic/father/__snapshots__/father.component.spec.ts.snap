// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FatherComponent debe hacer match con el snapshot 1`] = `
<div
  id="root1"
>
  <h1>
    Padre
  </h1>
  <button
    class="btn btn-dark"
  >
    Cargar cliente
  </button>
  <br />
  <br />
  <code
    class="mt-2 fs-5 fw-semibold"
  >
    
  </code>
  <hr />
  <app-father-son>
    <h3>
      Hijo
    </h3>
    <code
      class="mt-2 fs-5 fw-semibold"
    >
      
    </code>
    <br />
    <br />
    
    
  </app-father-son>
</div>
`;
